pkg/bugsnag/bugsnag.go

```diff
@@ -0,0 +1,360 @@
+	if !initialized {
+		_ = Initialize()
+	}
+
+	if !enabled || err == nil {
+		return
+	}
+
+	var c context.Context
+	if len(ctx) > 0 {
+		c = ctx[0]
+	} else {
+		c = context.Background()
+	}
+
+	_ = bugsnag.Notify(err, c, bugsnag.SeverityWarning)
+}
+
+// NotifyInfo reports informational events for monitoring and debugging purposes.
+// These are typically expected errors or noteworthy system events.
+func NotifyInfo(err error, ctx ...context.Context) {
+	if !initialized {
+		_ = Initialize()
+	}
+
+	if !enabled || err == nil {
+		return
+	}
+
+	var c context.Context
+	if len(ctx) > 0 {
+		c = ctx[0]
+	} else {
+		c = context.Background()
+	}
+
+	_ = bugsnag.Notify(err, c, bugsnag.SeverityInfo)
+}
+
+// Notify reports an error with custom severity level for flexible error categorization.
+// Use bugsnag.SeverityError, bugsnag.SeverityWarning, or bugsnag.SeverityInfo as severity values.
+func Notify(err error, severity interface{}, ctx ...context.Context) {
+	if !initialized {
+		_ = Initialize()
+	}
+
+	if !enabled || err == nil {
+		return
+	}
+
+	var c context.Context
+	if len(ctx) > 0 {
+		c = ctx[0]
+	} else {
+		c = context.Background()
+	}
+
+	_ = bugsnag.Notify(err, c, severity)
+}
+
+// NotifyWithMetadata reports an error with custom metadata for enhanced debugging context.
+// Additional metadata helps developers understand the state and conditions when errors occur.
+func NotifyWithMetadata(err error, severity interface{}, metadata bugsnag.MetaData, ctx ...context.Context) {
+	if !initialized {
+		_ = Initialize()
+	}
+
+	if !enabled || err == nil {
+		return
+	}
+
+	var c context.Context
+	if len(ctx) > 0 {
+		c = ctx[0]
+	} else {
+		c = context.Background()
+	}
+
+	_ = bugsnag.Notify(err, c, severity, metadata)
+}
+
+// WrapError enhances an error with additional context for better error tracking.
+// Use this to add descriptive information about where and why an error occurred.
+func WrapError(err error, message string) error {
+	if err == nil {
+		return nil
+	}
+	return fmt.Errorf("%s: %w", message, err)
+}
+
+// NotifyOnPanic captures and reports panic conditions before propagating them.
+// Always use with defer at the start of goroutines and main functions for comprehensive panic tracking.
+func NotifyOnPanic(ctx context.Context) {
+	if r := recover(); r != nil {
+		var err error
+		switch x := r.(type) {
+		case string:
+			err = fmt.Errorf("panic: %s", x)
+		case error:
+			err = fmt.Errorf("panic: %w", x)
+		default:
+			err = fmt.Errorf("panic: %v", r)
+		}
+
+		// Report panic as critical error
+		NotifyError(err, ctx)
+
+		// Preserve panic behavior for proper error handling
+		panic(r)
+	}
+}
+
+// Flush ensures all queued error reports are transmitted before application termination.
+// Call this in main() defer or before process exit to prevent error loss.
+func Flush() {
+	// Bugsnag Go client handles flushing internally through its async queue
+	// This function exists for API consistency and future enhancements
+}
```

Comment:
If this isn't doing anything don't include it 🙏 we can add it in future if we need it